hepcrawl/spiders/edp_spider.py

```diff
@@ -0,0 +1,430 @@
+        article_type = response.meta.get("article_type")
+        record = HEPLoader(item=HEPRecord(), selector=node, response=response)
+
+        record.add_value('dois', response.meta.get("dois"))
+        record.add_xpath('abstract', './/Abstract')
+        record.add_xpath('title', './/ArticleTitle/Title')
+        record.add_xpath('subtitle', './/ArticleTitle/Subtitle')
+        record.add_value('authors', self._get_authors_rich(node))
+        record.add_xpath('free_keywords', './/Subject/Keyword/text()')
+        journal_title = node.xpath(
+            './/JournalShortTitle/text()|//JournalTitle/text()').extract_first()
+        record.add_value('journal_title', journal_title)
+        record.add_xpath('journal_issue', './/Issue/text()')
+        record.add_xpath('journal_volume', './/Volume/text()')
+        fpage = node.xpath('.//FirstPage/text()').extract_first()
+        lpage = node.xpath('.//LastPage/text()').extract_first()
+        record.add_value('journal_fpage', fpage)
+        record.add_value('journal_lpage', lpage)
+        if fpage and lpage:
+            record.add_value('page_nr', str(int(lpage) - int(fpage) + 1))
+
+        published_date = self._get_date_published_rich(node)
+        record.add_xpath('journal_year', './/IssueID/Year/text()')
+        record.add_value('date_published', published_date)
+
+        record.add_xpath('copyright_holder', './/Copyright/text()')
+        record.add_value('collections', self._get_collections(
+            node, article_type, journal_title))
+
+        if "pdf_links" in response.meta:
+            record.add_value(
+                "additional_files",
+                self._create_fft_file(
+                    get_first(response.meta["pdf_links"]),
+                    "INSPIRE-PUBLIC",
+                    "Fulltext"
+                )
+            )
+        record.add_value("urls", response.meta.get("urls"))
+
+        return record.load_item()
+
+    def build_item_jats(self, response):
+        """Build the final HEPRecord with JATS-format XML ('jp')."""
+        node = get_node(response.meta["record"])
+        article_type = response.meta.get("article_type")
+
+        record = HEPLoader(item=HEPRecord(), selector=node, response=response)
+        if article_type in ['correction',
+                            'addendum']:
+            record.add_xpath('related_article_doi',
+                             ".//related-article[@ext-link-type='doi']/@href")
+            record.add_value('journal_doctype', article_type)
+
+        record.add_value('dois', response.meta.get("dois"))
+        record.add_xpath('page_nr', ".//counts/page-count/@count")
+        record.add_xpath('abstract', './/abstract[1]')
+        record.add_xpath('title', './/article-title/text()')
+        record.add_xpath('subtitle', './/subtitle/text()')
+        record.add_value('authors', self._get_authors(node))
+        record.add_xpath('collaborations', ".//contrib/collab/text()")
```

Comment:
I see you use a lot the add_xpath and add_value, maybe it would be nice to create a function that you can pass it just a dictionary (or even a set o args) and that it just calls add_xpath and/or add_value for each of them, for example:
```
# Instead of
record.add_xpath('something', './/something')
record.add_xpath('else', './/else')
# something like:
batch_add_xpath(record, something='.//something', else='.//else')
```

Not sure if it's worth it adding it now, but next time we have to add anything might be a nice extra.